backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/LdmlDeValidator.java

```diff
@@ -85,4 +99,81 @@ public void fatalError(SAXParseException exception) {
 
     return Norm.builder().document(result).build();
   }
+
+  /**
+   * Validate the norm against the schematron rules. Throws if a rule is not fulfilled.
+   *
+   * @param norm the norm to validate
+   * @throws IOException The schematron file couldn't be read
+   * @throws XmlProcessingException A problem occurred during the processing of the XML
+   * @throws LdmlDeSchematronException A Schematron rule is violated
+   */
+  public void validateSchematron(Norm norm) throws IOException {
+    Source xsltSource = new StreamSource(schematronXslt.getInputStream());
+    xsltSource.setSystemId(schematronXslt.getURL().toString());
+    Transformer transformer = null;
+    try {
+      transformer = new TransformerFactoryImpl().newTransformer(xsltSource);
+    } catch (TransformerConfigurationException e) {
+      throw new XmlProcessingException(e.getMessage(), e);
+    }
+
+    String xmlText = XmlMapper.toString(norm.getDocument());
```

Comment:
Why not just directly converting the `Document` to a `Source` --> `DomSource` to pass it to the transformer?

`Source xmlSource = new DOMSource(norm.getDocument())`